src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -257,8 +271,50 @@ public Instance(
         } else {
             log.info("collect_default_jvm_metrics is false - not collecting default JVM metrics");
         }
+
+        JMXbean = createJMXBean();
+    }
+
+    private ObjectName getObjName(String domain,String instance) throws MalformedObjectNameException {
+        //return new ObjectName(domain + ":name=" + name);
+        return new ObjectName(domain + ":target_instance=" + escape(instance));
+        //return new ObjectName(domain + ":name=" + ObjectName.quote(escape(instance)));
+        //return new ObjectName(domain, "name", escape(instance));
+    }
+
+    public String escape(String inputString){
+        final String[] metaCharacters = {"*","?"};
+    
+        for (int i = 0 ; i < metaCharacters.length ; i++){
+            if(inputString.contains(metaCharacters[i])){
+                inputString = inputString.replace(metaCharacters[i],"\\"+ metaCharacters[i]);
+            }
+        }
+        inputString = "\"" + inputString + "\"";
+        log.info("sanatized target_name is: " + inputString);
+        return inputString;
     }
 
+    private JmxfetchTelemetry createJMXBean(){
+        mbs =  ManagementFactory.getPlatformMBeanServer();
+        JmxfetchTelemetry bean = new JmxfetchTelemetry(this);
+        log.info("Created jmx bean for instance: " + this.getCheckName());
+
+     try {
+                JMXbean_name = getObjName("JMXFetch" , this.getName());
+                mbs.registerMBean(bean,JMXbean_name);
+                log.info("Succesfully registered jmx bean for instance: " + this.getCheckName());
+
+        } catch (MalformedObjectNameException | InstanceAlreadyExistsException | MBeanRegistrationException | NotCompliantMBeanException e) {
+                    log.warn("Could not register bean for instance: " + this.getCheckName());
```

Comment:
make sure you print which type of exception, I believe you're only printing out the stack trace at the moment